requirements.txt

```diff
@@ -8,6 +8,7 @@ certifi==2023.7.22
 charset-normalizer==3.2.0
 chromadb==0.4.11
 click==8.1.7
+duckdb==0.9.2
```

Comment:
duckdb is used to execute query over dataframe

Reply:
Why do we need this?

Reply:
For example, if the predicate is 'SELECT function1(x_min, y_min) FROM table WHERE function1(x_min, y_min) > 100', we need to run function1 first and get the result column 'function1_result'. Then we need to do a filter on the column 'function1_result'. duckdb is used to execute query 'SELECT function1_result FROM df WHERE function1_result > 100' over the dataframe. Do you think it's better to implement this by ourselves? For a normal function, I don't think we need to store results in a normal db. For ML model, if they register it as a bound service, the inference results will still be materialized into a normal db.

Reply:
I found we can directly do such operations by dataframe. I will modify it.

Reply:
Raise an issue